[doc][core][dashboard][state] Promote state api and dashboard usage in core user guide. (#35760)
ray-project/ray: doc/source/ray-core/tasks.rst

          for(int i = 0; i < 4; i++) {
            // This doesn't block.
            ray::Task(SlowFunction).Remote();
          a

> From Ray 2.0 onwards, you could also use :ref:`Ray's State API <state-api-overview-ref>``  to query the tasks, or go to the :ref:`Dashboard <observability-getting-started>`` to see more task details. (These features all require installing ray with `pip install "ray[default]"`). 

Reviewer: ```suggestion
From Ray 2.0 onwards, use the :ref:`State API <state-api-overview-ref>``  to query for information about the tasks, or view the :ref:`Dashboard <observability-getting-started>`` to see more task details. (These features require installing Ray with `pip install "ray[default]"`). 
```

.. code-block:: bash
  # This API is only available when you download Ray via `pip install "ray[default]"`
  ray list tasks

.. code-block:: bash

> # We could see there are 4 slow_function running, while my_function is finished.

Reviewer: ```suggestion
# The response shows 4 `slow_function` tasks and a finished task called `my_function`.
```

======== List: 2023-05-25 09:41:21.710772 ========

> Stats:

Reviewer 2: this part seems to be broken

          for(int i = 0; i < 4; i++) {
            // This doesn't block.
            ray::Task(SlowFunction).Remote();
          a

> From Ray 2.0 onwards, you can use the :ref:`State API <state-api-overview-ref>`  to query for information about tasks, or view them on the :ref:`Dashboard <observability-getting-started>`: 4 `slow_function` are running while `my_function` has already finished.

Reviewer 2: same as above

in :func:`ray.remote() <ray.remote>` and :meth:`.options() <ray.remote_function.RemoteFunction.options>`.
See :ref:`Ray fault tolerance <fault-tolerance>` for more details.

.. tip::

>     From Ray 2.3 onwards, you could also see the tasks exit details (e.g. OOM killed) and task retries attempts with increasing attempt number using :ref:`Ray's State API <state-api-overview-ref>``  to query the actors information, or go to the :ref:`Dashboard <observability-getting-started>`` Actor page to see more details. (These features all require installing ray with `pip install "ray[default]"`). 

Reviewer: ```suggestion
    From Ray 2.3 onwards, see the task exit details (for example, OOM killed) and task retry attempts with increasing attempt numbers using the :ref:`State API <state-api-overview-ref>``  to query for information about the actors, or see the Actors View in the :ref:`Dashboard <observability-getting-started>`` for more details. (These features require installing Ray with `pip install "ray[default]"`). 
```

>     From Ray 2.3 onwards, you could also see the tasks exit details (e.g. OOM killed) and task retries attempts with increasing attempt number using :ref:`Ray's State API <state-api-overview-ref>``  to query the actors information, or go to the :ref:`Dashboard <observability-getting-started>`` Actor page to see more details. (These features all require installing ray with `pip install "ray[default]"`). 

Reviewer 2: Can you add this to fault tolerance doc + embed the output to that doc (first vs second retry)?
